Unclean exit on database connection loss #68

anon8675309 opened the pull request:
When in non-batch mode, an exception can be thrown by the InfluxDB library.  This change will catch that exception and cause RuuviCollector to exit with a non-zero return code.

This defers handling the problem to the supervisory system (e.g., systemd).  It could restart the service any number of times, send off notifications, possibly restart the database service, or whatever is appropriate in each specific deployment.

Scrin commented:
Since some users may rely on the collector not exiting on temporary network disruptions, this behavior should be configurable (disabled by default). The new catch block and more meaningful logging is good to exist in either case, but the return should happen conditionally.

Perhaps `exitOnInfluxDBIOException=false` under the advanced section of the config would be appropriate for this

anon8675309 commented:
Good point about introducing unexpected changes in behavior.  I've updated the merge request with the option you suggested and changed main() to conditionally exit.

Scrin commented on src/main/java/fi/tkgwf/ruuvi/Main.java:
```diff
@@ -119,7 +123,9 @@ boolean run(final BufferedReader reader) {
         } catch (IOException ex) {
             LOG.error("Uncaught exception while reading measurements", ex);
-            return false;
+            if (Config.exitOnInfluxDBIOException()) {
```
Looks like you added this check around the wrong return statement, should be in the InfluxDBIOException catch block

anon8675309 replied:
Oops. :flushed: Fixed now
